cpnest/nest2pos.py

```diff
@@ -171,14 +171,17 @@ def autocorrelation(x):
     """
     Compute the autocorrelation of the chain
     """
-    N = len(x)
-    X=np.fft.fft(x-x.mean())
-    # We take the real part just to convert the complex output of fft to a real numpy float. The imaginary part if already 0 when coming out of the fft.
-    R = np.real(np.fft.ifft(X*X.conj()))
-    # Divide by an additional factor of 1/N since we are taking two fft and one ifft without unitary normalization, see: https://docs.scipy.org/doc/numpy/reference/routines.fft.html#module-numpy.fft
-    return R/N
-
-def acl(acf):
+    mean=x.mean()
+    var=np.var(x)
+    xp=x-mean
+
+    cf=np.fft.fft(xp)
+    sf=cf.conjugate()*cf
+    corr=np.fft.ifft(sf).real/var/len(x)
+
+    return corr
+
+def acl(acf, safety = 5.0):
     for i in range(len(acf)):
-        if acf[i] < i/5.0:
-            return acf[i]+1
+        if np.abs(acf[i]) < i/safety:
```

Comment:
The [autocorrelation length used in lalinference](https://git.ligo.org/lscsoft/lalsuite/-/blob/master/lalinference/lib/LALInferenceNestedSampler.c#L1261) is defined differently.
Let 
L = \sum_i^j 2*Cxx[i]
Where Cxx[i] is the autocorrelation at lag i for parameter x.
Then the autocorrelation length is the value of L when Cxx[j]<TOLERANCE, where TOLERANCE is 0.01 in LALInferenceNest.